### PR TITLE
[Inductor] Add test for Scheduler fusions

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89877
* __->__ #90014

Currently there is `test_vertical_fusion1` which fuses entirely during
the lowering stage and no buffers are realized. This adds
`test_scheduler_vertical_fusion1` which is the same test but with
several intermediate calculations realized so the scheduler is left
to do the fusion.

To support the test, this PR also adds:
- `metrics.ir_nodes_pre_fusion` which when compared with
`generated_kernel_count` tells us how many nodes were fused.
- `torch._test_inductor_realize` which is an identity operator in
eager, but under inductor also forces the input to be realized.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by jansel — APPROVED** [review]

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -12816,6 +12816,12 @@
   tags: view_copy
   autogen: _test_autograd_multiple_dispatch_view_copy.out
 
+# Note: this function is only for testing.
+- func: _test_inductor_realize(Tensor self) -> Tensor
```
Comment (ngimel) [inline aten/src/ATen/native/native_functions.yaml:12820]: I think we shouldn't be adding testing ops into pytorch proper, we can do python registration directly in the test instead, cc @SherlockNoMad

Comment (peterbell10) [reply aten/src/ATen/native/native_functions.yaml:12820]: Is it possible to register the backward/forward formulas from python?

Comment (ngimel) [reply aten/src/ATen/native/native_functions.yaml:12820]: Forward/backward can be handled with AutogradFunction, I don't remember though if we trace through it (which would work) or treat it as a leaf module (that won't).

**Pull request closed without merging**